Assets/Scripts/EnemyPatrol.cs

```diff
@@ -0,0 +1,239 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EnemyPatrol : MonoBehaviour
+{
+
+    [SerializeField]
+    private float moveSpeed, checkRadius, attackDelay, deathDelay, knockbackPower, knockbackHeight, knockbackTime;
+
+    public short enemyHealth;
+
+    private bool hittingWall, notEdge, playerInRange, moveRight, 
+        isAttacking, hasAttacked, attackedOnRight, canMove, isDamagable, beingKnockedBack;
+    
+    PlayerController playerController;
+
+    EnemyHealth EnemyHealthController;
+
+    CircleCollider2D enemyAttackPoint;
+
+    [SerializeField]
+    Transform pathCheck;
+
+    [SerializeField]
+    Transform edgeCheck;
+    
+    [SerializeField]
+    LayerMask whatIsWall, whatIsPlayer;
+    
+    Rigidbody2D enemyRigidBody;
+
+    Animator anim;
+
+    [SerializeField]
+    PhysicsMaterial2D knockbackPM, patrolPM;
+
+    Collider2D enemyPrimaryCollision;
+
+    // Use this for initialization
+    void Start() {
+        SetupObject();
+    }
+
+    //Gather and Set Necessary Components, GameObjects, Values, and States
```

Comment:
Change this comment to a summary tag. A lot of times you might call this type of function `Initialize()`